plutus-scb/src/Cardano/Wallet/Server.hs

```diff
@@ -39,72 +60,34 @@ import           Servant.Extra             (capture)
 -- block on access to a single, global 'MVar'.  We could do something
 -- smarter, but I don't think it matters as this is only a mock.
 asHandler ::
-       MVar State
-    -> LoggingT (StateT State (ExceptT ServantErr IO)) a
+    ClientEnv
+    -> MVar MockWalletState
+    -> Eff (AppEffects IO) a
     -> Handler a
-asHandler mVarState action =
+asHandler nodeClientEnv mVarState action =
     Handler $ do
         oldState <- liftIO $ takeMVar mVarState
-        (result, newState) <- runStateT (runStderrLoggingT action) oldState
-        liftIO $ putMVar mVarState newState
-        pure result
+        result <- liftIO $ runAppEffects nodeClientEnv oldState action
+        case result of
+            Left err -> MonadError.throwError $ err400 { errBody = Char8.pack (show err) }
```

Comment:
Shouldn't we put the original state back?

Reply:
Good catch! Oops!